src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -91,18 +100,23 @@ public boolean isLargeDiscrepancyBetweenMegaTag1And2(int limelightNumber) {
     PoseEstimate megaTag2Estimate = getMegaTag2PoseEstimate(limelightNumber);
 
     // Extract the positions of the two poses
-    Translation2d megaTag1Translation = megaTag1Estimate.pose.getTranslation();
-    Translation2d megaTag2Translation = megaTag2Estimate.pose.getTranslation();
+    Translation2d megaTag1TranslationMeters = megaTag1Estimate.pose.getTranslation();
+    Translation2d megaTag2TranslationMeters = megaTag2Estimate.pose.getTranslation();
+
+    double megaTag1RotationDegrees = megaTag1Estimate.pose.getRotation().getDegrees();
+    double megaTag2RotationDegrees = megaTag2Estimate.pose.getRotation().getDegrees();
 
     // Calculate the discrepancy between the two MegaTag translations in meters
-    double megaTagDiscrepancyMeters = megaTag1Translation.getDistance(megaTag2Translation);
+    double megaTagTranslationDiscrepancyMeters = megaTag1TranslationMeters.getDistance(megaTag2TranslationMeters);
+    double megaTagRotationDiscrepancyDegrees =  Math.abs(megaTag1RotationDegrees - megaTag2RotationDegrees);
 
     // Define a threshold (meters) for what constitutes a "large" discrepancy
     // This value should be determined based on your testing
     double thresholdMeters = 0.5;
+    double thresholdDegrees = 45;
 
     // Check if the discrepancy is larger than the threshold (meters)
-    return megaTagDiscrepancyMeters > thresholdMeters;
+    return megaTagTranslationDiscrepancyMeters > thresholdMeters || megaTagRotationDiscrepancyDegrees > thresholdDegrees;
```

Comment:
this is good

Reply:
We should try to set aside a whole meeting eventually where we just really make sure all of these values related to vision are good. So that also means really making sure all of our std devs are good, max rotation speed, etc.

Reply:
ok